java-source/src/test/java/net/corda/training/state/IOUStateTests.java

```diff
@@ -0,0 +1,217 @@
+        // Is the borrower field of the correct type?
+        assertTrue(borrowerField.getType().isAssignableFrom(Party.class));
+    }
+
+    /**
+     * Task 4.
+     * TODO: Add a 'paid' property of type [Amount] to the [IOUState] class to get this test to pass.
+     * Hint:
+     * - We would like this property to be initialised to a zero amount of Currency upon creation of the [IOUState].
+     * - You can use the [POUNDS] function from Currencies to create an amount of pounds e.g. 'Currencies.POUNDS(10)'.
+     * - This property keeps track of how much of the initial [IOUState.amount] has been settled by the borrower
+     *
+     * - We need to make sure that the [IOUState.paid] property is of the same currency type as the
+     *   [IOUState.amount] property. You can create an instance of the [Amount] class that takes a zero value and a token
+     *   representing the currency - which should be the same currency as the [IOUState.amount] property.
+     */
+    @Test
+    public void hasPaidFieldOfCorrectType() throws NoSuchFieldException {
+        // Does the paid field exist?
+        Field paidField = IOUState.class.getDeclaredField("paid");
+        // Is the paid field of the correct type?
+        assertTrue(paidField.getType().isAssignableFrom(Amount.class));
+    }
+
+    /**
+     * Task 5.
+     * TODO: Include the lender within the [IOUState.participants] list
+     * Hint: [ImmutableList.of] takes any number of parameters and will add them to the list
+     */
+    @Test
+    public void lenderIsParticipant() {
+        IOUState iouState = new IOUState(Currencies.POUNDS(0), ALICE.getParty(), BOB.getParty());
+        assertNotEquals(iouState.getParticipants().indexOf(ALICE.getParty()), -1);
+    }
+
+    /**
+     * Task 6.
+     * TODO: Similar to the last task, include the borrower within the [IOUState.participants] list
+     */
+    @Test
+    public void borrowerIsParticipant() {
+        IOUState iouState = new IOUState(Currencies.POUNDS(0), ALICE.getParty(), BOB.getParty());
+        assertNotEquals(iouState.getParticipants().indexOf(BOB.getParty()), -1);
+    }
+
+    /**
+     * Task 7.
+     * TODO: Implement [LinearState] along with the required methods.
+     * Hint: [LinearState] implements [ContractState] which defines an additional method. You can use
+     * IntellIJ to automatically add the member definitions for you or you can add them yourself. Look at the definition
+     * of [LinearState] for what requires adding.
+     */
+    @Test
+    public void isLinearState() {
+        assert(LinearState.class.isAssignableFrom(IOUState.class));
+    }
+
+    /**
+     * Task 8.
+     * TODO: Override the [LinearState.getLinearId()] method and assign it a value via your state's constructor.
+     * Hint:
+     * - [LinearState.getLinearId()] must return a [linearId] property of type [UniqueIdentifier]. You will need to create
+     * a new instance field.
```

Comment:
If we're going to use - as a bullet indicator then we should probably indent the text on subsequent lines so the bullets all stand out?

Reply:
This was the style used throughout all the Kotlin test code as well - I think for now it would be more consistent to leave it